DNN Platform/Library/Entities/Urls/AdvancedFriendlyUrlProvider.cs

```diff
@@ -443,38 +425,61 @@ private string FriendlyUrlInternal(TabInfo tab, string path, string pageName, st
                 //Get friendly path gets the standard dnn-style friendly path 
                 friendlyPath = GetFriendlyQueryString(tab, friendlyPath, pageName, localSettings);
 
-                if (portalSettings == null)
+                PortalSettings portalSettigsForFriendlyUrl = null;
+                if (cultureSpecificAlias && tab != null && portalSettings != null && portalAlias != portalSettings.PortalAlias.HTTPAlias)
                 {
-                    PortalAliasInfo alias = PortalAliasController.Instance.GetPortalAlias(portalAlias, tab.PortalID);
+                    // Gew new (culture specific) PortalAlias
+                    var pa = PortalAliasController.Instance.GetPortalAlias(portalAlias, portalSettings.PortalId);
+                    if (pa != null)
+                    {
+                        portalSettigsForFriendlyUrl = new PortalSettings(tab.TabID, pa);

```

Comment:
you create a new portal settings object but not use the value from parameters, this may cause unexpected result, such as if user modify some settings in portal settings object before call this method, those changes will lost.